libs/api/ebpf_api.cpp

```diff
@@ -2125,6 +2125,12 @@ _initialize_ebpf_programs_native(
             result = EBPF_NO_MEMORY;
             goto Exit;
         }
+        if (program->flags != 0) {
```

Comment:
Why the if test rather than always calling ebpf_program_set_flags?  Unless it's harmful, then always calling it means fewer code paths to test.

Reply:
This was done to match the libbpf behavior.